andela/hc-bigas-opus
# #155965580 Add multiple alert methods

# What does this PR do?

Setting up healthchecks to be able to send out important information concerning the cron jobs through multiple channels

## Description of Task to be completed?

Integrating several alert methods in the system such that the system is able to send a user alerts of all those selected platforms when a cron job is pinged and when it goes down.

## How should this be manually tested?

Open `health checks.io` on your localhost
Navigate to the integrations page.

* For email, click on the add integrations button in the email row, add the email you would like the notifications to be sent to. An email is sent to that email inbox to verify the setup. Click on the button in the email to confirm set up.

* For sms, click on the add integrations button in the sms row. You will be redirected to a page where you can add the phone number you would like to get your notifications on. Add the number and click on the save integration button. The number should be registered with twilio for it to work.

* For slack, click on the add integrations button in the sms row, you are redirected to a page where you 
can select the channel to which the notifications will be sent to from your list of available channels

* For Telegram, click on the add integrations button in the telegram row, you are redirected to a page. Add the callback url, then save it to get notifications

### The files changed/created include:

#### Created:
add_sms.html - Added UI for the page where the phone number is added
add_telegram.html - Added UI for the page where the telegram chat id is added
.env - Added the twilio configuration settings locally

#### Edited
hc/api/transports.py - Added the SMS model
hc/front/urls.py - Added the URL that redirects to the sms configuration page
hc/front/views.py - Added the function that loads the page where the number to sms is added
hc/templates/front/channels.html - Added sms to the list of configurable channel options
hc/front/tests/test_add_channel.py

What are the relevant pivotal tracker stories?
[#155965580](https://www.pivotaltracker.com/story/show/155965580)

#### Screenshots

##### Slack
<img width="994" alt="screen shot 2018-03-28 at 14 02 50" src="https://user-images.githubusercontent.com/15923109/38027483-79d9e2ce-3298-11e8-9340-45d0f9330385.png">

##### SMS
![screenshot_20180328-143214](https://user-images.githubusercontent.com/15923109/38027498-8742b1fc-3298-11e8-9d55-927fe723f83c.png)

##### Telegram
<img width="707" alt="screen shot 2018-03-29 at 00 46 36" src="https://user-images.githubusercontent.com/15923109/38058205-b6b7941e-32ea-11e8-9e99-75135fa38605.png">
